src/ShopDataProvider.php

```diff
@@ -44,4 +46,14 @@ public function getShopUrl(): string
 
         return $context->link->getBaseLink();
     }
+
+    /**
+     * Returns whether the shop opt-in to send its URL to the API
+     *
+     * @return bool
+     */
+    public function sendingShopUrlEnabled(): bool
```

Comment:
IMO this would fit better in `DistributionApi` and could be named `shouldAddShopUrl()`
Also, perhaps we could inject and use `prestashop.adapter.legacy.configuration` instead of Configuration